selfservice/hook/session_issuer.go

```diff
@@ -39,23 +39,27 @@ func NewSessionIssuer(r sessionIssuerDependencies) *SessionIssuer {
 	return &SessionIssuer{r: r}
 }
 
-func (e *SessionIssuer) ExecutePostRegistrationPostPersistHook(w http.ResponseWriter, r *http.Request, a *registration.Flow, s *session.Session) error {
+func (e *SessionIssuer) ExecutePostRegistrationPostPersistHook(w http.ResponseWriter, r *http.Request, a *registration.Flow, s *session.Session, continueWith *flow.ContinueWithContainer) error {
 	return otelx.WithSpan(r.Context(), "selfservice.hook.SessionIssuer.ExecutePostRegistrationPostPersistHook", func(ctx context.Context) error {
-		return e.executePostRegistrationPostPersistHook(w, r.WithContext(ctx), a, s)
+		return e.executePostRegistrationPostPersistHook(w, r.WithContext(ctx), a, s, continueWith)
 	})
 }
 
-func (e *SessionIssuer) executePostRegistrationPostPersistHook(w http.ResponseWriter, r *http.Request, a *registration.Flow, s *session.Session) error {
+func (e *SessionIssuer) executePostRegistrationPostPersistHook(w http.ResponseWriter, r *http.Request, a *registration.Flow, s *session.Session, c *flow.ContinueWithContainer) error {
 	s.AuthenticatedAt = time.Now().UTC()
 	if err := e.r.SessionPersister().UpsertSession(r.Context(), s); err != nil {
 		return err
 	}
 
 	if a.Type == flow.TypeAPI {
+		if c != nil {
+			c.Items = append(c.Items, NewContinueWith(s.Token))
+		}
 		e.r.Writer().Write(w, r, &registration.APIFlowResponse{
-			Session:  s,
-			Token:    s.Token,
-			Identity: s.Identity,
+			Session:      s,
+			Token:        s.Token,
+			Identity:     s.Identity,
+			ContinueWith: c.Items,
```

Comment:
This is essentially duplicated from https://github.com/ory/kratos/pull/3144/files#diff-2b4e8e6ce993f442aa73df0ab4d28c0dd56b66f3d5819908f96b965d73f8a3d0, but I didn't want to refactor that here, as that would also potentially change the order of these hooks.